src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -565,15 +565,23 @@ suggestFillHole :: Diagnostic -> [(T.Text, [TextEdit])]
 suggestFillHole Diagnostic{_range=_range,..}
     | Just holeName <- extractHoleName _message
     , (holeFits, refFits) <- processHoleSuggestions (T.lines _message)
-    = map (proposeHoleFit holeName False) holeFits
-    ++ map (proposeHoleFit holeName True) refFits
+    = let candidates = map (proposeHoleFit holeName False) holeFits
+                     ++ map (proposeHoleFit holeName True) refFits
+      in if length candidates > numberOfHolesLimit
+           then []
+           else candidates
     | otherwise = []
     where
       extractHoleName = fmap head . flip matchRegexUnifySpaces "Found hole: ([^ ]*)"
       proposeHoleFit holeName parenthise name =
           ( "replace " <> holeName <> " with " <> name
           , [TextEdit _range $ if parenthise then parens name else name])
       parens x = "(" <> x <> ")"
+      -- If there are more than this many holes, don't return any as they're
+      -- not likely to be specific enought to be useful
+      -- See https://github.com/haskell/haskell-language-server/issues/532
+      -- 5 chosen arbitrarily
+      numberOfHolesLimit = 5
```

Comment:
5 seems a bit low. I would go for something like 15. Can we make it user configurable?

Reply:
If it can be configured by ghc flags, then perhaps this is better left to cabal.project.local or something.

Reply:
But your limit, the one pepe asked for, is now different from the ghc flag, no? 
You can set 100 in the flag but you will not show suggestions if they are greater than 5.

Reply:
oh sure, it's slightly different; tbh I'd be happy with either

Reply:
I selected 5, because if the selection isn't instant, i.e. at a glance, then I'd be better off just typing what I want (I could certainly be blind to other's preferences or use-cases here!). Tbf I have absolutely no UX research behind this! 

[This](https://wordmaker.info/how-many/glance.html) site suggests that the "Total Number of words made out of [a] Glance = 50"